### DISCUSSION TITLE
could this be made compatible with the IPFS DHT?

### DISCUSSION THREAD

**Discussion by pedroapero** [bitmagnet-io/bitmagnet#147] [Ideas]
Hello,
I'm not sure about the differences between Bittorrent's DHT and IPFS DHT.

Would it be possible to make Bitmagnet index the IPFS DHT, that features data deduplication / reduces swarm fragmentation?

Thanks for all the work!

**Comment by mgdigital** [comment]
They make use of some similar algorithms but they're generally fairly different, and would require a completely separate implementation and domain model. So it's probably unlikely to happen. Here's my lazy answer: https://chat.openai.com/share/6e47c41e-c7fb-442e-88da-1e9b23aae154

**Reply by pedroapero** [reply]
The ChatGPT response is a total nonsense like often haha (differences highlighted are actually identical).

Thanks for your reply, I expected this to be complicated. From what I understand, the biggest difference is that one hash in the BT DHT corresponds to a full file or set of files; whereas in IPFS a hash can identify a folder or a piece of a larger hash (might be equivalent in Bittorrent v2).
